src/server/server_state.h

```diff
@@ -93,16 +93,33 @@ class ServerState {  // public struct - to allow initialization.
   void operator=(const ServerState&) = delete;
 
  public:
+  enum TxType { NORMAL, OOO, GLOBAL, NUM_TX_TYPES };
   struct Stats {
-    uint64_t ooo_tx_cnt = 0;
+    std::array<uint64_t, NUM_TX_TYPES> tx_type_cnt;
 
     uint64_t eval_io_coordination_cnt = 0;
     uint64_t eval_shardlocal_coordination_cnt = 0;
     uint64_t eval_squashed_flushes = 0;
 
     uint64_t tx_schedule_cancel_cnt = 0;
 
-    Stats& operator+=(const Stats& other);
+    // Array of size of number of shards.
+    // Each entry is how many transactions we had with this width (unique_shard_cnt).
+    uint64_t* tx_width_freq_arr = nullptr;
```

Comment:
we create a dynamically allocated array to count number of transactions per width. 
unfortunately, it required making stats moveable.

Reply:
Now when I think about it, could be cool having a `Moveable<T>` wrapper, that makes a pod type T moveable only, this would make it possible to define Stats as moveable using `= default `.

Reply:
Why do you manage memory manually, why not valarray (or vector)?

Reply:
Just seamed wasteful to keep size, capacity for run-time array that does not need to grow.

Reply:
It's 16 bytes per thread... 😄 (or 8 with fixed array)